src/views/blockDetails/index.tsx

```diff
@@ -1,89 +1,70 @@
 import { PDLink } from '@components/pdLink';
-import { ToggleButton } from '@components/toggleButton';
 import { useStoreChain } from '@stores';
 import { formatNumber } from '@utils/helpers';
 
+import { BlockBody } from './blockBody';
+import { BlockHeader } from './blockHeader';
 import styles from './styles.module.css';
 
-interface IBlockData {
-  number: number;
-  blockHash: string;
-  extrinsicsRoot: string;
-  parentHash: string;
-  stateRoot: string;
-  specVersion: number;
-  validatorId: string;
-  timeStamp: string;
-}
+import type { IMappedBlock } from '@custom-types/block';
 
 const BlockDetails = () => {
-  const latestFinalizedBlock = useStoreChain.use.finalizedBlock?.();
   const { blockNumber } = useParams();
+  const data = useStoreChain?.use?.blocksData?.();
+  const latestFinalizedBlock = useStoreChain.use.finalizedBlock?.();
 
-  const [isChecked, setIsChecked] = useState<boolean>(false);
-  const [blockData, setBlockData] = useState<IBlockData | undefined>();
-  const [isFinalized, setIsFinalized] = useState<boolean>(false);
-  const blocksData = useStoreChain?.use?.blocksData?.();
-
-  const handleSetCheck = useCallback((e: ChangeEvent<HTMLInputElement>) => {
-    const target = e.currentTarget.parentNode as HTMLDivElement;
-    const format = target.getAttribute('data-format');
-    target.setAttribute('data-format', format === 'utc' ? 'local' : 'utc');
-
-    setIsChecked(state => !state);
-  }, []);
+  const [blockData, setBlockData] = useState<IMappedBlock>();
 
   useEffect(() => {
-    if (blockNumber && blocksData.size > 0) {
-      const block = blocksData.get(Number(blockNumber));
+    if (blockNumber && data.size > 0) {
+      const block = data.get(Number(blockNumber));
       if (!block) {
         return;
       }
-      setBlockData({
-        number: block.header.number,
-        blockHash: block.header.hash,
-        extrinsicsRoot: block.header.extrinsicRoot,
-        parentHash: block.header.parentHash,
-        stateRoot: block.header.stateRoot,
-        specVersion: block.header.runtime?.spec_version || 0,
-        validatorId: block.header.identity.toString(),
-        timeStamp: format(new Date(block.header.timestamp), 'yyyy-MM-dd HH:mm:ss'),
-      });
-    }
-  }, [blockNumber, blocksData]);
 
-  useEffect(() => {
-    if (latestFinalizedBlock && blockData) {
-      const blockNumber = blockData.number || 0;
-      const isBlockFinalized = blockNumber <= latestFinalizedBlock;
+      if (block.header.number <= (latestFinalizedBlock ?? 0)) {
+        setBlockData({
+          ...block,
+          header: {
+            ...block.header,
+            isFinalized: true,
+          },
+        });
+        return;
+      }
 
-      setIsFinalized(isBlockFinalized);
+      setBlockData({
+        ...block,
+        header: {
+          ...block.header,
+          isFinalized: false,
+        },
+      });
     }
-  }, [latestFinalizedBlock, blockData]);
+  }, [blockNumber, data, latestFinalizedBlock]);
```

Comment:
What do you think about this one?

```suggestion
useEffect(() => {
    const block = data.get(Number(blockNumber));
    if (block) {
      setBlockData({
        ...block,
        header: {
          ...block.header,
          isFinalized: block.header.number <= (latestFinalizedBlock ?? 0),
        },
      });
    }
  }, [blockNumber, data, latestFinalizedBlock]);
 ```